app/react/ConnectionsList/components/ConnectionsGroup.js

```diff
@@ -43,39 +75,23 @@ export class ConnectionsGroup extends Component {
     this.setState({ selectedItems, selected: groupSelected });
   }
 
-  setGroupFilter(selectedItems) {
-    const newFilter = {};
-    newFilter[this.props.group.get('key')] = selectedItems;
-    this.props.setFilter(newFilter);
-  }
-
-  componentWillMount() {
-    this.setState({ expanded: true, selected: false, selectedItems: Immutable([]) });
-  }
+  toggleSelectGroup() {
+    const { group } = this.props;
+    const selectedItems = !this.state.selected
+      ? group.get('templates').map(i => group.get('key') + i.get('_id'))
+      : Immutable([]);
 
-  shouldComponentUpdate(nextProps, nextState) {
-    return (
-      !is(this.props.group, nextProps.group) ||
-      this.state.expanded !== nextState.expanded ||
-      this.state.selected !== nextState.selected ||
-      this.state.selectedItems.size !== nextState.selectedItems.size
-    );
+    this.setGroupFilter(selectedItems);
+    this.setState({ selected: !this.state.selected, selectedItems });
```

Comment:
I think we should follow this suggestion:
- since this.setState is changing the selected value, using the same state property can produce some unexpected results or unnecessary rerenders.  Using a callback which receives the current state is a much safer route.